File: solutions/leetcode_387.c
int firstUniqChar(char* s) {
    int chars[26] = {0};
    int i = 0;
    while (s[i] != '\0')
        chars[s[i++] - 'a']++;
    i = 0;
    while (s[i] != '\0')
        if (chars[s[i++] - 'a'] == 1)
            return i - 1;
    return -1;
}

File: solutions/leetcode_387.php
<?php

class Solution {

    /**
     * @param String $s
     * @return Integer
     */
    function firstUniqChar($s) {
        $chars = [];
        for ($i = 0; $i < strlen($s); $i++) {
            $chars[$s[$i]] = ($chars[$s[$i]] ?? 0) + 1;
        }
        for ($i = 0; $i < strlen($s); $i++) {
            if ($chars[$s[$i]] === 1) {
                return $i;
            }
        }
        return -1;
    }
}
